tools/setup

```diff
@@ -1,51 +1,98 @@
 #!/bin/sh
 
 ROOT=$(dirname $0)/..
-
 NVM_VERSION="v0.34.0"
-
-BAZEL_DEB="https://github.com/bazelbuild/bazel/releases/download/0.29.0/bazel_0.29.0-linux-x86_64.deb"
-BAZEL_PKG_NAME=$(basename "$BAZEL_DEB")
+BAZEL_VERSION="0.29.0"
 
 YLW="\033[33m"
 RED="\033[91m"
 GRN="\033[92m"
 MAG="\033[95m"
 BLD="\033[1m"
 END="\033[0m"
-
 CMD="$BLD$YLW"
 
+warn() {
+    echo "$RED$BLD$1$END"
+}
 
 fail() {
-    echo "$RED$BLD$1$END"
+    warn $1
     exit 1
 }
 
 status() {
     echo "$MAG$1$END"
 }
 
+# Detect Platform
+case $(uname | tr '[:upper:]' '[:lower:]') in
+  linux*)
+    OS_NAME=linux
+    BAZEL_DEP="https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel_$BAZEL_VERSION-installer-linux-x86_64.sh"
+    ;;
+  darwin*)
+    OS_NAME=osx
+    BAZEL_DEP="https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel-$BAZEL_VERSION-installer-darwin-x86_64.sh"
+    ;;
+  msys*)
+    OS_NAME=windows
+    BAZEL_DEP="https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel-$BAZEL_VERSION-windows-x86_64.exe"
+    # TODO(alxr) investigate windows compatability
+    fail "Windows is currently unsupported. Please follow install instructions in the README."
+    ;;
+  *)
+    OS_NAME=notset
+    warn "Can't detect host OS"
+    ;;
+esac
+
+BAZEL_PKG_NAME=$(basename "$BAZEL_DEP")
```

Comment:
PKG -> SCRIPT -- it's not actually installing a deb package anymore, it's running a script instead.

Reply:
Yes, that's what I intended. This seems to be the way the Bazel docs recommend: https://docs.bazel.build/versions/master/install-ubuntu.html

Reply:
Ah, are you saying update the var name? On it!